### PR TITLE
Fix several warnings

### PR DESCRIPTION
**Pull request by skyline75489**
Validation run: https://aiinfra.visualstudio.com/ONNX%20Runtime/_build/results?buildId=538132&view=results

### CODE REVIEW

File: .pipelines/stages/jobs/nuget-validation-job.yml [hunk 1]
```diff
@@ -165,7 +165,7 @@ jobs:
        cd examples/csharp/HelloPhi
        mv models/$(prebuild_phi3_mini_model_folder) models/phi-3
        dotnet restore -r $(os)-$(arch) /property:Configuration=$(csproj_configuration) --source https://api.nuget.org/v3/index.json --source https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/ORT-Nightly/nuget/v3/index.json --source $PWD --disable-parallel --verbosity detailed
-       dotnet build ./HelloPhi.csproj -r $(os)-$(arch) /property:Configuration=$(csproj_configuration)
+       dotnet build ./HelloPhi.csproj -r $(os)-$(arch) /property:Configuration=$(csproj_configuration) --no-restore --self-contained
```
Comment (skyline75489) [inline .pipelines/stages/jobs/nuget-validation-job.yml:168]: warning NETSDK1201: For projects targeting .NET 8.0 and higher, specifying a RuntimeIdentifier will no longer produce a self contained app by default

File: src/ort_genai_c.cpp [hunk 2]
```diff
@@ -242,7 +242,7 @@ OgaResult* OGA_API_CALL OgaGenerator_GetOutput(const OgaGenerator* oga_generator
               static_cast<uint8_t*>(ortvalue_output->GetTensorMutableRawData()) + data_size,
               static_cast<uint8_t*>(ortvalue_clone->GetTensorMutableRawData()));
   } else {
-    throw std::runtime_error("Unsupported Device type: " + ortvalue_output->GetTensorMemoryInfo().GetDeviceType());
+    throw std::runtime_error("Unsupported Device type: " + std::to_string(ortvalue_output->GetTensorMemoryInfo().GetDeviceType()));
```
Comment (skyline75489) [inline src/ort_genai_c.cpp:245]: Clang `-Wstring-plus-int`

File: src/search.h [hunk 3]
```diff
@@ -93,7 +93,7 @@ struct BeamSearch_Cpu : Search_Cpu {
   RoamingArray<int32_t> GetSequence(size_t index) override;
   RoamingArray<int32_t> GetSequence(size_t batch_id, size_t beam_id);
 
-  bool IsDone() const;
+  bool IsDone() const override;
```
Comment (skyline75489) [inline src/search.h:96]: Clang/GCC `-Winconsistent-missing-override`

**Review by RyanUnderhill — APPROVED** [review]
Looks good, I'm surprised the override one slipped through.

**Pull request merged**